plugins/genesys/fu-genesys-usbhub-firmware.c

```diff
@@ -240,197 +297,16 @@ fu_genesys_usbhub_firmware_parse(FuFirmware *firmware,
 	return TRUE;
 }
 
-static GByteArray *
-fu_genesys_usbhub_firmware_write(FuFirmware *firmware, GError **error)
```

Comment:
i actually think we need these; else we can't convert the genesys-usbhub.builder.xml to the binary file for the fuzzer to use

Reply:
I'm not sure the usage of [_write](https://github.com/fwupd/fwupd/blob/main/docs/tutorial.md#write) and [_build](https://github.com/fwupd/fwupd/blob/main/docs/tutorial.md#build). Because there are many wrong set before. And the firmware blob shall not change if it was signed. 

To fulfill all ICs, the work may too heavy. Or can I revert the functions and keep wrong there?

Reply:
> I'm not sure the usage of _write

They're basically to get a file that's close enough to reality so that the parse function can be fuzzed. I don't think they have to be perfect, and you probably can just revert the removal for it to work.

Reply:
OK, only `_write`?  what about `_build`?

Reply:
It's needed to parse anything in genesys-usbhub.builder.xml -- i think it should remain tbh.

Reply:
I reverted the functions and tested GL3525 to export and build, it's failed. The flows are:
1. fwupdtool firmware-export GL3525.bin genesys-usbhub > 3525.xml
2. remove the loading line in 3525.xml
3. fwupdtool firmware-build 3525.xml 3525.bin >> show `Unsupported IC` by [here](https://github.com/fwupd/fwupd/pull/5851/files#diff-63c9be4ca40269945c2bcd376b55bdf11e87c9fd975b50e336d48bd5a84c0ab2R104)

I removed the `_write()` and repeat the flows, it's success. Did I use the wrong flow to do the test?

Reply:
Don't worry about parsing the thing that you're exporting -- write() is trying to give the fuzzer a headstart on the file format. Of course, being able the parse the thing you exported is probably a good thing to have from a testing point of view.